File: 372. Super Pow/C++/Solution.cpp
class Solution {
public:
    int superPow(int a, vector<int> &b) {
        class Utils {
        public:
            int pow(int a, int n, int base) {
                int result = 1;
                a %= base;
                for (int i = 0; i < n; i++) {
                    result = (result * a) % base;
                }
                return result;
            }
        };
        if (b.empty()) {
            return 1;
        }
        int last = b.back();
        b.pop_back();
        int base = 1337;
        Utils utils;
        return utils.pow(superPow(a, b), 10, base) * utils.pow(a, last, base) % base;
    }
};

File: 372. Super Pow/Python/Solution.py
class Solution(object):
    def superPow(self, a, b):
        """
        :type a: int
        :type b: List[int]
        :rtype: int
        """
        def pow(a, n, base):
            result = 1
            for _ in range(n):
                result = (result * a) % base
            return result

        if not b:
            return 1
        last = b.pop()
        base = 1337
        return pow(self.superPow(a, b), 10, base) * pow(a, last, base) % base
